go/database/mpt/rlp/rlp.go

```diff
@@ -55,7 +55,11 @@ func EncodeInto(dst []byte, item Item) []byte {
 func Decode(rlp []byte) (Item, error) {
 	item, consumed, err := decode(rlp)
 	if consumed < uint64(len(rlp)) {
-		return nil, fmt.Errorf("unexpected trailing data: %x", rlp[consumed:])
+		for _, b := range rlp[consumed:] {
```

Comment:
A line of comment justifying trailing zeros here would be useful. I assume it is due to the padding of embedded nodes?

Reply:
np, added